custom_components/sat/coordinator.py

```
@@ -268,14 +308,27 @@ async def async_will_remove_from_hass(self) -> None:
         """Control the heating loop for the device."""
         current_time = datetime.now()
-
-        # Make sure we have valid value
+        last_boiler_temperature = self.boiler_temperatures[-1][1] if self.boiler_temperatures else None
+
+        # Check and handle boiler temperature tracking
+        if last_boiler_temperature is not None:
+            if not self.flame_active:
+                self._tracking_boiler_temperature = True
+            elif self._tracking_boiler_temperature:
+                if self.setpoint > self.boiler_temperature and self.setpoint - 3 < self.boiler_temperature < last_boiler_temperature:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)